flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -134,7 +136,31 @@ public void registerScalarFunction(String name, ScalarFunction function) {
 	}
 
 	public String[] getUserDefinedFunctions() {
-		List<String> result = new ArrayList<>();
+		String[] names = getUserDefinedFunctionNames().toArray(new String[0]);
+		Arrays.sort(names);
+
+		return names;
+	}
+
+	public String[] getFunctions() {
+		Set<String> result = getUserDefinedFunctionNames();
+
+		// Get built-in functions
+		result.addAll(
+			BuiltInFunctionDefinitions.getDefinitions()
+				.stream()
+				.map(f -> normalizeName(f.getName()))
+				.collect(Collectors.toSet())
+		);
+
+		String[] names = result.toArray(new String[0]);
+		Arrays.sort(names);
```

Comment:
I'm not sure if we need to sort the returned array. (same for getUserDefinedFunctions()). If a caller needs a sorted result, it can do on its own. This is also for consistency, as we are not sorting other listX().

For this matter, I think it makes sense for sql-client to show a sorted list. In this case, sql-client can sort and show. However, other clients may not need a sorted list.